packages/stark-rbac/src/modules/authorization/directives/hide-on-permission.directive.ts

```diff
@@ -0,0 +1,83 @@
+import { Directive, Inject, Input, OnDestroy, OnInit, TemplateRef, ViewContainerRef, ViewRef } from "@angular/core";
+import { STARK_LOGGING_SERVICE, StarkLoggingService } from "@nationalbankbelgium/stark-core";
+import { StarkRBACDirectivePermission } from "./permission.intf";
+import { STARK_RBAC_AUTHORIZATION_SERVICE, StarkRBACAuthorizationService } from "../services/authorization.service.intf";
+
+/**
+ * Name of the directive
+ */
+const directiveName: string = "[starkHideOnPermission]";
+
+/**
+ * [Structural Directive](https://angular.io/guide/structural-directives#structural-directives) to remove an element if the user has any of the roles specified
+ * in the {@link StarkRBACDirectivePermission} object passed as input.
+ *
+ * This directive should be used in cases where some content in the application should be shown to all the users except to those that have a specific role.
+ *
+ * @example
+ * <div *starkHideOnPermission="permissionsObject">
+ *   <!-- protected content to be shown only to those users that don't have any of the specified roles -->
+ *   Some protected content
+ * </div>
+ */
+@Directive({
+	selector: directiveName
+})
+export class StarkHideOnPermissionDirective implements OnInit, OnDestroy {
```

Comment:
As we discussed the other day, using getters/setters is a more practical approach to handle the inputs changes than the `ngOnChanges` as long as they are atomic, meaning that there are no inter-dependencies between the different `@Input`s... like in this case 😉  Otherwise, in this particular case the code will be a bit more verbose with no added value in my opinion...

I do agree to have `ngOnChanges` when the inputs are related to each other like in the `StarkTableComponent`

Reply:
My bad, I didn't pay enough attention to the setter. So you're right! No need to implement `OnChanges` 😊

Reply:
About this... it is mentioned in the Angular docs: https://angular.io/guide/component-interaction#intercept-input-property-changes-with-ngonchanges